### ISSUE TITLE
Convergence of training method

### ISSUE THREAD

**Issue by jmren168** [zc8340311/RobustAutoencoder#5]
Hi there,

I would like to plot the convergence of ADMM training method, but am not sure what I should do. 

In "l21RobustDeepAutoencoderOnSt.py", 
        for it in range(iteration):
            if verbose:
                print("Out iteration: " , it)
            ## alternating project, first project to L
            self.L = X - self.S
            ## Using L to train the auto-encoder
            self.AE.fit(self.L, sess = sess,
                        iteration = inner_iteration,
                        learning_rate = learning_rate,
                        batch_size = batch_size,
                        init = re_init,
                        verbose = verbose)
            ## get optmized L
            self.L = self.AE.getRecon(X = self.L, sess = sess)
            ## alternating project, now project to S and shrink S
            self.S = SHR.l21shrink(self.lambda_, (X - self.L).T).T

            **WRITE some codes to compute the objective function min||self.L - D(E(self.L))||2 + lambda||self.S||2,1 HERE?**

Any comments will be appreciated.

**Comment by zc8340311** [comment]
Hi, 
Thanks for your interests. I had code testing its convergence in some older versions, but I can not find it.
I will write down some codes for monitoring convergence. But I can not guarantee delivering it very quickly.

The first term can be computed by:
1. get recon of autoencoder: recon = self.AE.getRecon(X = self.L, sess=sess)
2. compute two norm of  first_term = np.linalg.norm(self.L - recon) 
 
The second term is quite easy:
sec_term = lam * np.sum( np.linalg.norm(self.S, axis=0))

The ob function for each iteration is:
ob = first_term + sec_term

**Comment by jmren168** [comment]
Hi, 

I tried to write down codes, in "l21RobustDeepAutoencoderOnST.py".

![image](https://user-images.githubusercontent.com/31725302/41568547-c9ec239e-7398-11e8-9904-fee6851e6cc7.png)

The first term here I wrote as np.linalg.norm(L_ - self.L) because I thought it should be projected L minus optimized L. Correct me if I'm wrong, and thanks for your kindly response.

Best,
JM

**Comment by zc8340311** [comment]
Hi JM,
I think you are correct. It represents how the autoencoder part get trained in every iteration.
Sorry for late response and many thanks for your comments.

Sincerely,
Chong
